include/ignition/math/Region3.hh

```diff
@@ -0,0 +1,200 @@
+      /// \return the x-axis interval
+      public: const Interval<T> &Ix() const { return this->ix; }
+
+      /// \brief Get the y-axis interval for the region
+      /// \return the y-axis interval
+      public: const Interval<T> &Iy() const { return this->iy; }
+
+      /// \brief Get the z-axis interval for the region
+      /// \return the z-axis interval
+      public: const Interval<T> &Iz() const { return this->iz; }
+
+      /// \brief Check if the region is empty
+      /// A region is empty if any of the intervals
+      /// it is defined with (i.e. Ix, Iy, Iz) are.
+      /// \return true if it is empty, false otherwise
+      public: bool Empty() const
+      {
+        return this->ix.Empty() || this->iy.Empty() || this->iz.Empty();
+      }
+
+      /// \brief Check if the region contains `_point`
+      /// \param[in] _point point to check for membership
+      /// \return true if it is contained, false otherwise
+      public: bool Contains(const Vector3<T> &_point) const
+      {
+        return (this->ix.Contains(_point.X()) &&
+                this->iy.Contains(_point.Y()) &&
+                this->iz.Contains(_point.Z()));
+      }
+
+      /// \brief Check if the region contains `_other` region
+      /// \param[in] _other region to check for membership
+      /// \return true if it is contained, false otherwise
+      public: bool Contains(const Region3<T> &_other) const
+      {
+        return (this->ix.Contains(_other.ix) &&
+                this->iy.Contains(_other.iy) &&
+                this->iz.Contains(_other.iz));
+      }
+
+      /// \brief Check if the region intersects `_other` region
+      /// \param[in] _other region to check for intersection
+      /// \return true if it is contained, false otherwise
+      public: bool Intersects(const Region3<T>& _other) const
+      {
+        return (this->ix.Intersects(_other.ix) &&
+                this->iy.Intersects(_other.iy) &&
+                this->iz.Intersects(_other.iz));
+      }
+
+      /// \brief Equality test operator
+      /// \param _other region to check for equality
+      /// \return true if regions are equal, false otherwise
+      public: bool operator==(const Region3<T> &_other) const
+      {
+        return this->Contains(_other) && _other.Contains(*this);
+      }
+
+      /// \brief Inequality test operator
+      /// \param _other region to check for inequality
+      /// \return true if regions are unequal, false otherwise
+      public: bool operator!=(const Region3<T> &_other) const
+      {
+        return !this->Contains(_other) || !_other.Contains(*this);
+      }
+
+      /// \brief Stream insertion operator
+      /// \param _out output stream
+      /// \param _r Region3 to output
+      /// \return the stream
+      public: friend std::ostream &operator<<(
+        std::ostream &_out, const ignition::math::Region3<T> &_r)
+      {
+        return _out <<_r.ix << " x " << _r.iy << " x " << _r.iz;
```

Comment:
I haven't found any references for the notation of multi-dimensional intervals. Is this some well-known notation? I'm wondering whether instead of `I x I x I`, we should use something like `I I I`, `I, I, I`, `I ✕ I ✕ I` or something else.

Reply:
> Is this some well-known notation?

@chapulina I wouldn't call it well-known, no. Having said that, I can't think of a simpler notation than a cartesian product of coordinate intervals. I used `x` instead of `✕` to keep unicode out of terminals, but perhaps that's a non-issue.

Reply:
Thanks for the clarification. I'm ok with this since I don't think there's another standard we can use instead.